### PR TITLE
Fixed out of resources problem 

### PR DESCRIPTION
**Pull request by ducbueno**
Fixed the out of resources problem (issue #2754).

There are still a lot of time step chopping.

### CODE REVIEW

**Comment by blattms** [comment]
:tada: Great, I am really impressed. Will test right away.

Would you mind telling what the problem was (and maybe include that info in the git commit message via `git commit --amend` and a `git push --force`)? Was it not storing the cl::Program, yet using kernels later?

**Comment by ducbueno** [comment]
The problem was that the OpenCL context, queue and kernel were being set in the `initialize` function, which is called only in the `solve_system` function. For some reason the `wellContribs` object gets destroyed after the first `solve_system` run and, because of this, the context, queue and kernel previously set ceased to exists, along with all the buffers from `wellContribs`, causing the out of resources error.

**Review by blattms — CHANGES_REQUESTED** [review]
I just would like to seen the workgroup size changed.

This indeed fixes the memory problem and unfortunately reveils the next bug (time step chopping even for SPE1). Great step in the right direction, though. Happy to merge when the above is addressed,

File: opm/simulators/linalg/bda/openclSolverBackend.cpp [hunk 1]
```diff
@@ -71,7 +71,7 @@ unsigned int openclSolverBackend<block_size>::ceilDivision(const unsigned int A,
 template <unsigned int block_size>
 double openclSolverBackend<block_size>::dot_w(cl::Buffer in1, cl::Buffer in2, cl::Buffer out)
 {
-    const unsigned int work_group_size = 256;
+    const unsigned int work_group_size = 1024;
```
[the remark this answers is not in the document]
Comment (ducbueno) [reply opm/simulators/linalg/bda/openclSolverBackend.cpp:74]: Yes, not problem changing it back. I was just running some tests.

**Comment by blattms** [comment]
skipping jenkins as it ignores the GPU stuff.

**Review by blattms — APPROVED** [review]

**Comment by akva2** [comment]
I have installed opencl libraries on jenkins, so you get build testing now. No driver, and thus still no runtime testing.

Debian 10 does not supply cuda so that remains in the dark.

**Comment by blattms** [comment]
> Debian 10 does not supply cuda so that remains in the dark.

Mine does:
```
mblatt@smaug:~$ apt-cache policy nvidia-cuda-dev
nvidia-cuda-dev:
  Installiert:           9.2.148-7
  Installationskandidat: 9.2.148-7
  Versionstabelle:
     10.1.243-6~bpo10+1 100
        100 http://deb.debian.org/debian buster-backports/non-free amd64 Packages
 *** 9.2.148-7 500
        500 http://deb.debian.org/debian buster/non-free amd64 Packages
        100 /var/lib/dpkg/status
mblatt@smaug:~$
```

But cuda is messy as it demands gcc-7

**Pull request merged**